lua/gitblame/init.lua

```diff
@@ -621,61 +645,79 @@ local function clear_all_extmarks()
     end
 end
 
--- this function is uesed to verify the config info for debounce
---- @return boolean
-local verify_debounce_info = function()
-    if vim.g.gitblame_schedule_event ~= "CursorMoved" and vim.g.gitblame_schedule_event ~= "CursorHold" then
-        vim.notify(
-            string.format("event is error: %s, it just can be CursorMoved or CursorHold", vim.g.gitblame_schedule_event),
-            vim.log.levels.ERROR,
-            {}
-        )
-        return false
-    end
-    if vim.g.gitblame_clear_event ~= "CursorMovedI" and vim.g.gitblame_clear_event ~= "CursorHoldI" then
-        vim.notify(
-            string.format("event is error: %s, it just can be CursorMoved or CursorHold", vim.g.gitblame_clear_event),
-            vim.log.levels.ERROR,
-            {}
+-- Validates the `parameter_name` against `available_values`. Returns an error message
+-- if `parameter_name` doesn't match `available_values`.
+--
+---@param parameter_name string
+---@param available_values object[]
+---@return string|nil
+local function validate_enum_parameter(parameter_name, available_values)
+    local current_value = vim.g[parameter_name]
+
+    if not vim.tbl_contains(available_values, current_value) then
+        return string.format(
+            "Invalid value for `%s`: %s. Available values are %s",
+            parameter_name,
+            current_value,
+            vim.inspect(available_values)
         )
-        return false
     end
+end
+
+-- Verifies the debounce configuration and displays an error message if it is invalid.
+-- Returns `true` if the configuration is valid, `false` otherwise.
+--
+---@return boolean
+local function verify_debounce_configuration()
+    local error_message = validate_enum_parameter("gitblame_schedule_event", { "CursorMoved", "CursorHold" })
+        or validate_enum_parameter("gitblame_clear_event", { "CursorMovedI", "CursorHoldI" })
 
     if type(vim.g.gitblame_delay) ~= "number" or vim.g.gitblame_delay < 0 then
-        vim.notify(
-            string.format("delay is error: %s, it just can be number", vim.g.gitblame_delay),
-            vim.log.levels.ERROR,
-            {}
-        )
-        return false
+        error_message =
+            string.format("Invald value for `gitblame_delay`: %s. It should be a positive number", vim.g.gitblame_delay)
+    end
+
+    if error_message ~= nil then
+        vim.notify(error_message, vim.log.levels.ERROR, {})
+    end
+
+    return error_message == nil
+end
+
+---@type function
+local function maybe_clear_virtual_text_and_schedule_info_display()
+    local position_info = get_position_info()
+
+    if not position_info.is_on_same_line and not need_update_after_horizontal_move then
+        clear_virtual_text()
     end
 
-    return true
+    debounce(schedule_show_info_display, math.floor(500))()
```

Comment:
Did you mean to use `gitblame_delay` here? I don't know how configurable this should be - `500` works fine as a static value but I also think it makes sense to use the delay prop so you can make it longer or shorter if needed (0 makes the virtual text never leave, 1000 makes it delay even longer).

```suggestion
    debounce(schedule_show_info_display, math.floor(vim.g.gitblame_delay))()
```

Reply:
Oh, this was some testing code, thanks!